extensions/wikia/Recirculation/js/helpers/DataHelper.js

```diff
@@ -3,38 +3,21 @@ define('ext.wikia.recirculation.helpers.data', [
 	'jquery',
 	'wikia.window',
 	'wikia.abTest',
-	'wikia.nirvana'
-], function ($, w, abTest, nirvana) {
+	'wikia.nirvana',
+	'ext.wikia.recirculation.utils'
+], function ($, w, abTest, nirvana, utils) {
 
-	return function() {
+	return function(options) {
+		options = options || {};
 
 		function loadData() {
-			return $.when(loadAll(), loadArticles())
-				.done(function(data, articles) {
-					data.articles = articles;
-					return data;
-				});
-		}
-
-		function loadAll() {
-			var deferred = $.Deferred();
-
-			nirvana.sendRequest({
-				controller: 'RecirculationApi',
-				method: 'getAllPosts',
-				format: 'json',
-				type: 'get',
-				scriptPath: w.wgCdnApiUrl,
-				data: {
-					cityId: w.wgCityId
-				},
-				callback: function(data) {
-					data = formatFandom(data);
-					deferred.resolve(data);
-				}
-			});
-
-			return deferred.promise();
+			switch(options.type) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement with "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1301)